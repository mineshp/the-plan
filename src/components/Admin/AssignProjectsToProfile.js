import React from 'react';
import PropTypes from 'prop-types';
import { Button, Checkbox, Container, Form, Select } from 'semantic-ui-react';

const AssignProjectsToProfile = ({
    handleSubmitToAssignProjectsToProfile,
    handleProjectCheckboxChange,
    handleDropDownChange,
    profiles,
    projects
}) => {
    const DropDownProfileItems = [];
    profiles.map((profile) => DropDownProfileItems.push({
        key: profile._id, // eslint-disable-line no-underscore-dangle
        value: profile.name,
        text: profile.name.toUpperCase()
    }));

    const CheckBoxItems = [];
    projects.map((project) => {
        // eslint-disable-next-line no-underscore-dangle
        const projectId = project._id;

        return CheckBoxItems.push(
            <Form.Field key={projectId}>
                <Checkbox
                    label={project.projectName}
                    id={projectId}
                    data-projcb={project}
                    onChange={handleProjectCheckboxChange}
                />
            </Form.Field>
        );
    });

    return (
        <Container>
            <Form onSubmit={handleSubmitToAssignProjectsToProfile}>
                <Form.Field>
                    <Select
                        placeholder="Select Profile"
                        className="form-dropdown"
                        options={DropDownProfileItems}
                        onChange={handleDropDownChange}
                    />
                </Form.Field>
                {CheckBoxItems}
                <Button color="green" type="submit">Save</Button>
            </Form>
        </Container>
    );
};

AssignProjectsToProfile.propTypes = {
    handleDropDownChange: PropTypes.func.isRequired,
    handleSubmitToAssignProjectsToProfile: PropTypes.func.isRequired,
    handleProjectCheckboxChange: PropTypes.func.isRequired,
    profiles: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string.isRequired,
        active: PropTypes.bool.isRequired
    })).isRequired,
    projects: PropTypes.arrayOf(PropTypes.shape({
        _id: PropTypes.string.isRequired,
        projectName: PropTypes.string.isRequired,
    })).isRequired
};

export default AssignProjectsToProfile;
