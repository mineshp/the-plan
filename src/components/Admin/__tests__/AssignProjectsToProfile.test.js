import React from 'react';
import renderer from 'react-test-renderer';
import AssignProjectsToProfile from '../AssignProjectsToProfile';
import { mockListProfiles } from '../../../helpers/test/testData/controlCentreData';
import { mockListProjects } from '../../../helpers/test/testData/projectData';

const mockHandleSubmitToAssignProjectsToProfile = jest.fn();
const mockHandleDropDownChange = jest.fn();
const mockHandleProjectCheckboxChange = jest.fn();

describe('Admin > Manage > AssignProjectsToProfile', () => {
    const mockProfiles = mockListProfiles();
    const mockProjects = mockListProjects();
    it('renders ListUsers component', () => {
        const tree = renderer.create(
            <AssignProjectsToProfile
                profiles={mockProfiles}
                projects={mockProjects}
                handleSubmitToAssignProjectsToProfile={mockHandleSubmitToAssignProjectsToProfile}
                handleDropDownChange={mockHandleDropDownChange}
                handleProjectCheckboxChange={mockHandleProjectCheckboxChange}
            />
        );
        expect(tree).toMatchSnapshot();
    });
});
