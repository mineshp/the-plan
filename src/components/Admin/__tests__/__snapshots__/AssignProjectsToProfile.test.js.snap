// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin > Manage > AssignProjectsToProfile renders ListUsers component 1`] = `
<div
  className="ui container"
>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <div
        aria-busy={undefined}
        aria-disabled={undefined}
        aria-expanded={false}
        aria-multiselectable={undefined}
        className="ui selection dropdown form-dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="default text"
          role="alert"
        >
          Select Profile
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            aria-checked={false}
            aria-disabled={undefined}
            aria-selected={true}
            className="selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              MIN_JIGS_WEDDING
            </span>
          </div>
          <div
            aria-checked={false}
            aria-disabled={undefined}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              HOME_DIY
            </span>
          </div>
          <div
            aria-checked={false}
            aria-disabled={undefined}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              CHRISTMAS
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui checkbox"
        data-projcb={
          Object {
            "_id": "5992d50066a7043f2598e12d",
            "colour": "red",
            "createdDate": "2017-08-15T12:02:00.000Z",
            "profilesAssigned": Array [],
            "projectDescription": "a description",
            "projectName": "Iron Man",
          }
        }
        id="5992d50066a7043f2598e12d"
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          name={undefined}
          readOnly={true}
          tabIndex={0}
          type="checkbox"
          value={undefined}
        />
        <label>
          Iron Man
        </label>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui checkbox"
        data-projcb={
          Object {
            "__v": 0,
            "_id": "59a08710e57cb1da97cd1477",
            "colour": "red",
            "createdDate": "2017-08-25T20:22:40.994Z",
            "profilesAssigned": Array [],
            "projectDescription": "a description",
            "projectName": "Thor",
          }
        }
        id="59a08710e57cb1da97cd1477"
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          name={undefined}
          readOnly={true}
          tabIndex={0}
          type="checkbox"
          value={undefined}
        />
        <label>
          Thor
        </label>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui checkbox"
        data-projcb={
          Object {
            "__v": 0,
            "_id": "59a08bb6e57cb1da97cd1478",
            "colour": "grey",
            "createdDate": "2017-08-25T20:42:30.159Z",
            "profilesAssigned": Array [],
            "projectDescription": "a description",
            "projectName": "Captain America",
          }
        }
        id="59a08bb6e57cb1da97cd1478"
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          name={undefined}
          readOnly={true}
          tabIndex={0}
          type="checkbox"
          value={undefined}
        />
        <label>
          Captain America
        </label>
      </div>
    </div>
    <button
      className="ui green button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
      type="submit"
    >
      Save
    </button>
  </form>
</div>
`;
